lib/flop/schema.ex

```diff
@@ -1219,51 +1221,77 @@ defimpl Flop.Schema, for: Any do
   end
 
   # credo:disable-for-next-line
-  def build_cursor_dynamic_func_normal do
-    quote do
-      def cursor_dynamic(_, [{direction, field}], cursor) do
-        field_cursor = cursor[field]
+  def build_cursor_dynamic_func_normal(fields) do
+    for field <- Enum.uniq(fields) do
+      quote do
+        def cursor_dynamic(_, [{direction, unquote(field)}], %{
+              unquote(field) => field_cursor
+            })
+            when not is_nil(field_cursor) and
+                   direction in [:asc, :asc_nulls_first, :asc_nulls_last] do
+          dynamic(
+            [r],
+            field(r, ^unquote(field)) >
+              type(^field_cursor, field(r, unquote(field)))
```

Comment:
We probably need to apply the same changes to `build_cursor_dynamic_func_join`, at least if the field is configured with the `ecto_type` option?

Reply:
Probably yes, I add this